### PR TITLE
Don't rsync checkpoint_tmp directories.

### LINKED ISSUE #18427: [tune] checkpoint_tmp directories break Analysis.get_best_checkpoint
### What is the problem?

When training on a cluster there are spurious `checkpoint_tmpxxxxxx` directories left on head node when tuning completes. This never happens when training on head node only.

These `checkpoint_tmpxxxxxx` directories break `Analysis.get_best_checkpoint`. Example stacktrace:

```
Traceback (most recent call last):
  ...
  File "/home/max/src/research/es/tune_results.py", line 42, in <lambda>
    df['model_dir'] = df['logdir'].map(lambda d: a.get_best_checkpoint(d).rstrip('/'))
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/ray/tune/analysis/experiment_analysis.py", line 303, in get_best_checkpoint
    checkpoint_paths = self.get_trial_checkpoints_paths(trial, metric)
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/ray/tune/analysis/experiment_analysis.py", line 268, in get_trial_checkpoints_paths
    chkpt_df = TrainableUtil.get_checkpoints_paths(trial_dir)
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/ray/tune/utils/trainable.py", line 170, in get_checkpoints_paths
    chkpt_iter = int(chkpt_dir[chkpt_dir.rfind("_") + 1:])
ValueError: invalid literal for int() with base 10: 'tmp17ae56'
```

I wonder why these `checkpoint_tmpxxxxxx` directories come into existence and don't get removed automatically?

*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray: ray-2.0.0.dev0
Python: 3.8.10
OS: Ubuntu 20.04 LTS

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by max0x7ba**
## Why are these changes needed?

checkpoint_tmpxxxxxx directories must not be synced from the worker nodes to the head node. 

## Related issue number

Closes #18427 
@richardliaw 

## Checks

I tested the change with my distributed tuning in Google Compute Engine using list-of-nodes cluster mode.

No ray unittest can run on my workstation for some reason.

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/syncer.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ def log_sync_template(options=""):
 
     rsh = "ssh -i {ssh_key} -o ConnectTimeout=120s -o StrictHostKeyChecking=no"
     rsh = rsh.format(ssh_key=quote(ssh_key))
+    options += " --exclude='checkpoint_tmp*'"
```
Comment (max0x7ba) [inline python/ray/tune/syncer.py:86]: May be there should be one project-wide rsync-filter file for checkpoints because rsync looks to be used elsewhere.

**Review by xwjiang2010 — APPROVED** [review]
Thanks for the change!

**Comment by max0x7ba** [comment]
> Thanks for the change!

Thank you for accepting my change.

Thinking more about it, it would be ideal if all temporary files/directories in `~/ray_results` directory followed the same temporary file/directory naming convention, e.g. the common `*~` (e.g. `checkpoint_tmpxxxxxx~`), so that this one exclude pattern for `rsync` filters out all and any temporary files and directories, including the ones added in the future, as long as they follow the same naming convention.

**Comment by xwjiang2010** [comment]
I see. Yeah I think that makes sense. I am noting it down. We will probably refactor some checkpointing logic for the next quarter. @krfricke @matthewdeng

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Comment by krfricke** [comment]
Sorry for letting this slip. I just merged master to kick off CI once again and will then merge.

**Review by krfricke — APPROVED** [review]

File: python/ray/tune/syncer.py [hunk 2]
```diff
@@ -129,6 +129,8 @@ def log_sync_template(options: str = ""):
 
     rsh = "ssh -i {ssh_key} -o ConnectTimeout=120s -o StrictHostKeyChecking=no"
     rsh = rsh.format(ssh_key=quote(ssh_key))
+
+    options += " --exclude='checkpoint_tmp*'"
```
Comment (krfricke) [inline python/ray/tune/syncer.py:133]:
```suggestion
    options += " --exclude='checkpoint_tmp*'"
```

**Pull request merged**